tests/test_all_fields.py

```diff
@@ -0,0 +1,187 @@
+import os
+import unittest
+
+from functools import reduce
+from singer import metadata
+import tap_tester.connections as connections
+import tap_tester.menagerie   as menagerie
+import tap_tester.runner      as runner
+
+from base import TestSquareBase
+from test_client import TestClient
+
+
+class TestSquareAllFields(TestSquareBase):
+    """Test that with all fields selected for a stream we replicate data as expected"""
+
+    START_DATE = ""
+
+    def name(self):
+        return "tap_tester_square_all_fields"
+
+    def testable_streams(self):
+        return set(self.expected_streams()).difference(
+            {'employees', 'locations'} # STREAMS THAT CANNOT CURRENTLY BE TESTED
+        )
+
+    @classmethod
+    def setUpClass(cls):
+        print("\n\nTEST SETUP\n")
+        cls.client = TestClient()
+
+    @classmethod
+    def tearDownClass(cls):
+        print("\n\nTEST TEARDOWN\n\n")
+
+    def test_run(self):
+        """
+        Verify that for each stream you can get data when no fields are selected
+        and only the automatic fields are replicated.
+        """
+
+        print("\n\nRUNNING {}\n\n".format(self.name()))
+
+        # ensure data exists for sync streams and set expectations
+        self.START_DATE = self.get_properties().get('start_date')
+        expected_records = {x: [] for x in self.expected_streams()} # ids by stream
+        for stream in self.testable_streams():
+            existing_objects = self.client.get_all(stream, self.START_DATE)
+            if existing_objects:
+                print("Data exists for stream: {}".format(stream))
+                for obj in existing_objects:
+                    expected_records[stream].append(obj)
+            else:
+               print("Data does not exist for stream: {}".format(stream))
+               assert None, "more test functinality needed"
```

Comment:
Is this asserting that None is not equal to a string? Shouldn't it be a fail("message")?

Reply:
This is a placeholder, too early to implement a create for all testable streams in this test. And it is not necessary since data exists for all streams under test. By the end each test will have methods to ensure that if data does not exist, it is created prior to the test execution. That way we ensure each test is independent of existing data and of other tests.

Reply:
Oh I misread your comment no, it is asserting None so that the test fails. And then the output is `AssertionError: more test functionality needed`

Reply:
Right, I'm saying https://docs.python.org/3/library/unittest.html#unittest.TestCase.fail is more clear than `assert None, message`.

Reply:
Not a big deal